### DISCUSSION TITLE
PWA - precache big vendor.js - how to configure?

### DISCUSSION THREAD

**Discussion by robert-hanuschke** [quasarframework/quasar#8940] [CLI - PWA mode] [answered]
Hello,

I'm trying to configure my PWA to precache a rather big vendor.js. When I execute `quasar build -m pwa`, I get the warning:

> App · ⚠️  1 warning encountered:
>/js/vendor.686378d5.js is 2.12 MB, and won't be precached. Configure maximumFileSizeToCacheInBytes to change this limit.

Looking through the Quasar and workbox documentations, I modified my quasar.conf.js to include:

```
const {InjectManifest} = require('workbox-webpack-plugin');
module.exports = function (/* ctx */) {
    return {
        ...
        build: {
            ...
            extendWebpack (cfg) {
                cfg.plugins.push(new InjectManifest({swSrc: './src-pwa/custom-service-worker.js', maximumFileSizeToCacheInBytes: 3000000, }))
            },
```

Unfortunately, this does not change the warning I get.

Can you please tell me how to configure this correctly?

Kind regards,
Robert

**Comment by metalsadman** [comment, marked as answer]
@robert-hanuschke in `quasar.conf.js>pwa>workboxOptions` https://quasar.dev/quasar-cli/developing-pwa/configuring-pwa#quasar-conf-js.
You don't need your manual setup above, just choose `workboxPluginMode: 'InjectManifest'`.

**Reply by robert-hanuschke** [reply]
Hello Aldrin,

seems I misinterpreted the documentation in a way that workboxOptions only works for GenerateSW and not for InjectManifest as a whole. ("The first one generates a service worker automatically, based on quasar.conf.js > pwa > workboxOptions (if any), while the second mode allows you to write your own service worker file.").

But after adding my maximumFileSizeToCacheInBytes there, the warning disappearad.

Thank you for the quick response and great help!

**Reply by jitendra1607** [reply]
@robert-hanuschke Can you help me with custom-service-worker.js file, after upgrading 1.* to 2.* I am facing errors.

https://github.com/quasarframework/quasar/discussions/9048
